react/features/base/tracks/actions.js

```diff
@@ -795,6 +787,51 @@ export function destroyLocalDesktopTrackIfExists() {
     };
 }
 
+/**
+ * Create an action for when a screenshare track's muted state has been signaled to be changed.
+ *
+ * @param {(JitsiLocalTrack|JitsiRemoteTrack)} track - JitsiTrack instance.
+ * @returns {Function}
+ */
+export function screenshareTrackMutedChanged(track) {
```

Comment:
I've updated this function to directly handle creating/removing fake SS participants instead of going through the middleware

Reply:
Please see my comment above, `screenshareTrackMutedChanged` and `screenshareVideoTypeChanged` essentially do the same thing, either dispatch a `createFakeScreenShareParticipant` event or a `participantLeft` event based on whether the videoType has changed or a if the mute state of a desktop track has been updated. Do you think these can be handled in the middleware ?

Reply:
We could but I guess it felt a bit weird that a track event triggers a change to the participant state. I generally use middlewares for logging and fetching external data. I think the advantages of having a centralized place (tracks middleware) makes it easier to control the creation of fake SS

Reply:
Yes, moving the creation of fake SS to tracks middleware will make it easier to maintain but I see what you are saying w.r.t debugging issues. Let's keep it the way it is now for this PR. We can revisit this later if the need arises. @paweldomas do you agree ?

Reply:
I haven't worked in this codebase long enough so I do want your opinions on what is the best approach with the tradeoffs in mind. Since this feature is still fresh in our heads, it might be easier to make the change now rather than later